src/together/resources/finetune.py

```
@@ -174,6 +174,18 @@ def create_finetune_request(
             f"training_method must be one of {', '.join(AVAILABLE_TRAINING_METHODS)}"
         )
 
+    if train_on_inputs is not None and training_method != "sft":
+        raise ValueError("train_on_inputs is only supported for SFT training")
+
+    if train_on_inputs is None and training_method == "sft":
+        log_warn_once(
+            "train_on_inputs is not set for SFT training, it will be set to 'auto' automatically"
+        )
+        train_on_inputs = "auto"
+
+    if dpo_beta is not None and training_method != "dpo":
```

Other option might be just a warning. What do you think?

---

good question! i don't have a strong opinion here, but i am leaning towards the error because non-None value means the user intentionally set it. if they intentionally set it, it might be because they assumed dpo would be active. so we should cancel and let them know that it isn't, rather than silently continuing with an sft job. wdyt @artek0chumak ?